tools/coredump/new.go

```diff
@@ -195,32 +199,34 @@ func (cmd *newCmd) exec(context.Context, []string) (err error) {
 	return nil
 }
 
-func dumpCore(pid uint64) (string, error) {
-	// Backup current coredump filter mask.
-	// https://man7.org/linux/man-pages/man5/core.5.html
-	coredumpFilterPath := fmt.Sprintf("/proc/%d/coredump_filter", pid)
-	prevMask, err := os.ReadFile(coredumpFilterPath)
-	if err != nil {
-		return "", fmt.Errorf("failed to read coredump filter: %w", err)
-	}
-	// Adjust coredump filter mask.
-	//nolint:gosec
-	err = os.WriteFile(coredumpFilterPath, []byte("0x3f"), 0o644)
-	if err != nil {
-		return "", fmt.Errorf("failed to write coredump filter: %w", err)
-	}
-	// Restore coredump filter mask upon leaving the function.
-	defer func() {
+func dumpCore(pid uint64, noModuleBundling bool) (string, error) {
+	if noModuleBundling {
+		// Backup current coredump filter mask.
+		// https://man7.org/linux/man-pages/man5/core.5.html
+		coredumpFilterPath := fmt.Sprintf("/proc/%d/coredump_filter", pid)
+		prevMask, err := os.ReadFile(coredumpFilterPath)
+		if err != nil {
+			return "", fmt.Errorf("failed to read coredump filter: %w", err)
+		}
+		// Adjust coredump filter mask.
 		//nolint:gosec
-		err2 := os.WriteFile(coredumpFilterPath, prevMask, 0o644)
-		if err2 != nil {
-			log.Warnf("Failed to restore previous coredump filter: %v", err2)
+		err = os.WriteFile(coredumpFilterPath, []byte("0x3f"), 0o644)
+		if err != nil {
+			return "", fmt.Errorf("failed to write coredump filter: %w", err)
 		}
-	}()
+		// Restore coredump filter mask upon leaving the function.
+		defer func() {
+			//nolint:gosec
+			err2 := os.WriteFile(coredumpFilterPath, prevMask, 0o644)
```

Comment:
We should convert prevMask to a hex string and write that to coredump_filter otherwise we'll run into the following:

```
WARN[0000] Failed to restore previous coredump filter: write /proc/367787/coredump_filter: invalid argument
```

Or just from the shell:

```
$ cat /proc/367787/coredump_filter 
00000033
$ echo 00000033 >/proc/367787/coredump_filter
$ cat /proc/367787/coredump_filter 
0000001b
$ echo 0000001b >/proc/367787/coredump_filter
-bash: echo: write error: Invalid argument
```

Reply:
Strange kernel inconsistency. And this is just code I reindented. But added a fix by just prepending a `0x` literal in the front.

Reply:
Thanks, this fixes it.